api/api.go

```diff
@@ -25,6 +26,60 @@ type PublicController struct {
 	Database *sql.DB
 }
 
+func (f *PublicController) GetIdFromCookie(c *gin.Context) int {
+	Cookie := c.Request.Header.Get("Cookie")
+	if Cookie == "" {
+		log.Errorf("Unable to get cookie")
+		c.IndentedJSON(http.StatusUnauthorized, "Unable to get cookie")
+		return -1
+	}
+	tokenString := strings.Split(Cookie, "=")[1]
+
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		return []byte("foobar"), nil
+	})
+	if err != nil {
+		log.Errorf("Error parsing token: %s\n", err.Error())
+		return -1
+	}
+
+	data := token.Claims.(jwt.MapClaims)["data"].(map[string]interface{})
+	id, err := strconv.Atoi(data["id"].(string))
+	if err != nil {
+		log.Errorf("Unable to convert idstring to int: %s\n", err.Error())
+		return -1
+	}
+
+	return id
+}
+
+func (f *PublicController) GetRoleIdFromCookie(c *gin.Context) int {
```

Comment:
Hier das gleiche wie in `GetIdFromCookie`

Reply:
Done